### ISSUE TITLE
Enable Replication of Configuration Data

### ISSUE THREAD

**Issue by michmike** [goharbor/harbor#7993]
Today we enable project replication and we replicate the artifacts and data. However, some users are interested in replicating the config as well (policies, users, RBAC, etc). One particular user that asked for this is Dell Tech that is hosting harbor internally as a service provider.

**Comment by alexvmw** [comment]
@michmike can you put me in touch w/ dell guys?  can potentially replicate quota policies, tag retentions policies, webhooks json payloads as well

**Comment by michmike** [comment]
yes, we should replicate all policy that would be needed to set up an identical project

**Comment by xaleeks** [comment]
from user @dandydeveloper

Is your feature request related to a problem? Please describe.
I did a quick search for this, I'm sure it's been requested before but I couldn't find anything. Please close if that's the case (sorry).

It would be good if the replication supported additional objects from the Harbor database to replicate over other settings like:

    Users
    Robot Accounts
    Policies
    Labels

Etc. Etc.

Right now, I have a number of Harbor instances that I replicate against and having to recreate and reconfigure everytime can be painful. This would also be a great option for DR scenarios.

Describe the solution you'd like
I think the summary describes the request well enough.

Describe the main design/architecture of your solution
I haven't looked into the code, but I guess if we add an adapter of some kind that says "If selected repository is Harbor, add options for Harbor settings" then through the backend processing of the request, add a sync cron of some kind that replicates over these settings via the Harbor API / RPC (Do we have RPC???) over the selected protocol.

Describe the development plan you've considered
I will update this when I get a closer look at the areas this would affect. Right now, I don't have a good solution.

Additional context
This should ultimately piggyback on the existing replication logic, this specific use case is for Harbor -> Harbor replicatio
